executor/code_loader.py

```diff
@@ -0,0 +1,112 @@
+import time
+import importlib
+import sys
+import signal
+import studentapi
+from studentapi import CodeExecutionError, _print
+import typing
+import studentcode
+
+
+def sigalarm_handler(signum, frame):
+    """Handler function that is called when process is interrupted by a SIGALRM. Causes process to cleanly exit."""
+    raise TimeoutError
+
+# To register the handler for the specified signal
+signal.signal(signal.SIGALRM, sigalarm_handler)
```

Comment:
This will set a signal handler even if you just import the module. We should avoid hidden side effects by moving the registration under the `__name__ == '__main__'` guard.

Reply:
Hey, I should've deleted `code_loader.py` before, but this was just the initial version I tried load the code. I don't use this file anymore and it's now deprecated so I'm going to delete it now.  All the logic for calling the student code in threads is now in `executor.c` if you want to check that out. Same thing for the rest of the comments on `code_loader` so I'm going to resolve them.

Reply:
OK, yeah, that's what I figured. I don't know enough about Python's C API to help.